src/main/java/fr/insalyon/creatis/gasw/executor/batch/internals/BatchJob.java

```diff
@@ -0,0 +1,187 @@
+     * 
+     * @throws GaswException
+     */
+    public void prepare() throws GaswException {
+        final RemoteTerminal rt = new RemoteTerminal(data.getConfig());
+
+        rt.connect();
+        for (final RemoteFile file : data.getFilesUpload()) {
+            log.info("Uploading file : " + file.getSource());
+            rt.upload(file.getSource(), file.getDest());
+        }
+        rt.disconnect();
+    }
+
+    /**
+     * Download all the data created by the jobs (not the app output) but the logs.
+     */
+    public void download() {
+        final RemoteTerminal rt = new RemoteTerminal(data.getConfig());
+
+        try {
+            rt.connect();
+            for (final RemoteFile file : data.getFilesDownload()) {
+                log.info("Downloading file : " + file.getSource());
+                rt.download(file.getSource(), file.getDest());
+            }
+            rt.disconnect();
+
+        } catch (GaswException e) {
+            log.error("Failed to download the files !", e);
+        }
+    }
+
+    public void submit() throws GaswException {
+        final BatchEngines engine = data.getConfig().getOptions().getBatchEngine();
+        final RemoteCommand command = engine.getSubmit(data.getWorkingDir() + data.getJobID() + ".batch");
+
+        try {
+            command.execute(data.getConfig());
+
+            if (command.failed()) {
+                throw new GaswException("Command failed !");
+            }
+            data.setBatchJobID(command.result());
+            log.debug("Job ID inside the Cluster : " + command.result());
+
+        } catch (GaswException e) {
+            log.error("Failed to submit the job " + getData().getJobID());
+            throw e;
+        }
+    }
+
+    public void start() throws GaswException {
+        prepare();
+        createBatchFile();
+        submit();
+        setStatus(GaswStatus.SUCCESSFULLY_SUBMITTED);
+    }
+
+    private GaswStatus convertStatus(final String status) {
+        if (status == null) {
+            return GaswStatus.UNDEFINED;
+        }
+        switch (status) {
+            case "COMPLETE":
+                return GaswStatus.COMPLETED;
+            case "COMPLETED":
+                return GaswStatus.COMPLETED;
+            case "PENDING":
+                return GaswStatus.QUEUED;
+            case "CONFIGURING":
+                return GaswStatus.QUEUED;
+            case "RUNNING":
+                return GaswStatus.RUNNING;
+            case "FAILED":
+                return GaswStatus.ERROR;
+            case "NODE_FAIL":
+                return GaswStatus.ERROR;
+            case "BOOT_FAIL":
+                return GaswStatus.ERROR;
+            case "OUT_OF_MEMORY":
+                return GaswStatus.ERROR;
+            default:
+                return GaswStatus.UNDEFINED;
+        }
+    }
+
+    private GaswStatus getStatusRequest() {
+        final BatchEngines engine = data.getConfig().getOptions().getBatchEngine();
```

Comment:
add a method that does `getConfig().getOptions().getBatchEngine()` directly in BatchJobData to simplify that.